### PR TITLE
backend/fix: #372 added gender in profile apis

### PR DESCRIPTION
**Pull request by soumyajit2304**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added Gender response in Profile API
Added Gender update in Profile API
Added Unmasked Email in Profile API

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by 0utkarsh** [comment]
Can you add migrations?

**Comment by soumyajit2304** [comment]
> Can you add migrations?

Gender column already exits in the person table

**Review by 0utkarsh — APPROVED** [review]

**Review by roman-bodavskiy — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs [hunk 1]
```diff
@@ -124,6 +125,7 @@ updatePerson personId req = do
       mbEncEmail
       (req.deviceToken)
       (req.language)
+      (req.gender)
```
Comment (roman-bodavskiy) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs:128]: redundant brackets

**Review by piyushKumar-1 — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Person.hs [hunk 2]
```diff
@@ -61,7 +61,7 @@ instance ToHttpApiData IdentifierType where
   toHeader = BSL.toStrict . encode
 
 --------------------------------------------------------------------------------------------------
-data Gender = MALE | FEMALE | OTHER | UNKNOWN
+data Gender = MALE | FEMALE | OTHER | UNKNOWN | PREFERNOTTOSAY
```
Comment (DonDokidon) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Person.hs:64]: use snake case

**Review by DonDokidon — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Person.hs (hunk 2 above)
Comment (soumyajit2304) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Person.hs]: Done

**Review by hkmangla — APPROVED** [review]

**Pull request merged**